src/utils/components/PendingChanges/utils/helpers.ts

```diff
@@ -143,8 +154,72 @@ export const getChangedNics = (vm: V1VirtualMachine, vmi: V1VirtualMachineInstan
   )
     return [];
 
-  return changedNics;
+  return changedNICs;
+};
+
+export const hasPendingChange = (pendingChange: PendingChange[]) =>
+  pendingChange?.[0]?.hasPendingChange;
+
+// Checks for other types of changes and non-hot-plug NIC changes
+export const nonHotPlugNICChangesExist = (
+  pendingChanges: PendingChange[],
+  nonHotPlugNICsExist: boolean,
+) => {
+  const moreChangeTypesExist = pendingChanges?.every(
+    (change) => change?.tabLabel === VirtualMachineDetailsTabLabel.NetworkInterfaces,
+  );
+  return moreChangeTypesExist || nonHotPlugNICsExist;
+};
+
+const getSortedNICs = (
+  vm: V1VirtualMachine,
+  vmi: V1VirtualMachineInstance,
+): { hotPlugNICs: string[]; nonHotPlugNICs: string[] } => {
+  const changedNICs = getChangedNICs(vm, vmi);
+  return changedNICs?.reduce(
+    (acc, nicName) => {
+      const isHotPlug = Boolean(getInterfaceByName(nicName, vm, vmi)?.bridge);
+      isHotPlug ? acc.hotPlugNICs.push(nicName) : acc.nonHotPlugNICs.push(nicName);
+      return acc;
+    },
+    { hotPlugNICs: [], nonHotPlugNICs: [] },
+  );
+};
+
+export const getSortedNICPendingChanges = (vm, vmi, history) => {
+  const sortedNICs = getSortedNICs(vm, vmi);
+  const { hotPlugNICs, nonHotPlugNICs } = sortedNICs;
+
+  return {
+    hotPlugPendingChanges: [
+      {
+        handleAction: () => {
+          history.push(getTabURL(vm, VirtualMachineDetailsTab.NetworkInterfaces));
+        },
+        hasPendingChange: !isEmpty(hotPlugNICs),
+        label:
+          !isEmpty(hotPlugNICs) && hotPlugNICs?.length > 1
```

Comment:
isn't just checking hotPlugNICs?.length > 1 is enough?
same in line 214 this file